product_pricelist_margins/models/product_pricelist_item.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2024 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+from odoo.tools.float_utils import float_is_zero
+
+
+class ProductPricelistItem(models.Model):
+    _inherit = "product.pricelist.item"
+
+    cost = fields.Float(
+        related="product_tmpl_id.standard_price",
+        digits="Product Price",
+    )
+    margin = fields.Float(
+        compute="_compute_margin",
+        digits="Product Price",
+    )
+    margin_percent = fields.Float(
+        string="Margin (%)",
+        compute="_compute_margin",
+    )
+
+    @api.depends(
+        "compute_price",
+        "applied_on",
+        "percent_price",
+        "base",
+        "price_discount",
+        "price_surcharge",
+        "price_round",
+        "price_min_margin",
+        "product_tmpl_id",
+        "cost",
+    )
+    def _compute_margin(self):
+        for item in self.filtered(
+            lambda x: x.applied_on in ("1_product", "0_product_variant")
+            and x.compute_price == "fixed"
```

Comment:
```suggestion
```

I don't understand this line. The algorithm can work for all compute_price method. don't you think ?